src/design-pages/passkey_kerberos.rst

```diff
@@ -0,0 +1,223 @@
+by the libfido2 library.
+
+Use cases
+---------
+As a user, not only do I want to walk to any system on a network and log in
+(desktop and/or terminal) using a passkey device connected to the system, but I
+also want to get a Kerberos ticket to identify myself to other services.
+
+Prerequisites
+-------------
+To register a new
+`pre-authentication data identifier <https://www.iana.org/assignments/kerberos-parameters/kerberos-parameters.xhtml#pre-authentication>`__
+in Kerberos for passkey it would be good to first finalize the IdP plugin
+design and security analysis requested by this community.
+
+Solution overview
+-----------------
+The objective is to locally authenticate and issue a Kerberos ticket to a
+centrally managed user. For that purpose SSSD will need to be modified to
+accommodate the new workflow, while still allowing the old one, where the user
+was authenticated locally without issuing any Kerberos ticket. The new workflow
+will be used for online authentication against an IPA server. The old workflow
+will be used for any offline authentication and for AD and other LDAP servers.
+
+SSSD will send a pre_auth message to the Kerberos client. It will send an
+AS-REQ message to the Kerberos KDC that will return an error because the
+assertion data is missing. Thus, SSSD will obtain the assertion data from
+the passkey and it will send the assertion to the Kerberos client, which will
+relay it to the KDC server. This in turn will relay the request to the
+ipa-otpd, which will retrieve the user information and ask the passkey_child
+to validate the assertion. Once validated the Kerberos KDC will issue a ticket
+and send it to SSSD via the Kerberos client.
+
+.. image:: ../images/passkey_kerberos_1.png
+  :alt: Architecture diagram
+
+Implementation details
+----------------------
+
+General overview
+****************
+When SSSD support for passkey authentication is enabled, a Kerberos client
+would advertise a support for passkey during AS-REQ request. If information
+about the Kerberos principal on the KDC side allows use of passkey
+authentication, KDC would respond with a pre-authentication data to choose
+passkey authentication.
+
+The following sequence diagram illustrates the workflow taking into account all
+the components involved in it:
+
+.. image:: ../images/passkey_kerberos_2.png
+  :alt: Sequence diagram
+
+* The user will try to login with a centrally managed account in the system.
+
+* PAM will relay this request to SSSD’s PAM responder.
+
+* The PAM responder will retrieve the user information from the cache, which
+  will provide this information including the authentication protocol (PASSKEY).
+
+* The PAM responder will try to pre_authenticate the user with PASSKEY.
+
+* The IPA provider will execute the krb5_child, which in turn will communicate
+  with the Kerberos client.
+
+* Kerberos client will send an AS-REQ with the PASSKEY pre-authentication data.
+
```

Comment:
Kerberos KDC will receive an AS-REQ and will query Kerberos database driver (KDB) for the information about the Kerberos principal. It will find out that the principal object supports pass key pre-authentication.

Reply:
Fixed